server/data/data-postgres/src/main/java/org/apache/james/domainlist/postgres/PostgresDomainList.java

```diff
@@ -41,7 +41,7 @@ public class PostgresDomainList extends AbstractDomainList {
     private final PostgresExecutor postgresExecutor;
 
     @Inject
-    public PostgresDomainList(DNSService dnsService, @Named(DEFAULT_INJECT) PostgresExecutor postgresExecutor) {
+    public PostgresDomainList(DNSService dnsService, @Named(POOL_INJECT_NAME) PostgresExecutor postgresExecutor) {
```

Comment:
My feeling is this "pool" semantic change here and everywhere else is not needed and can be avoided.

Maybe we can modify `SinglePostgresConnectionFactory` to be backed by the pool connection (and rename it), then avoid the `PoolPostgresExecutor` introduction?

We already have this code:
```java
    @Provides
    @Singleton
    JamesPostgresConnectionFactory provideJamesPostgresConnectionFactory(PostgresConfiguration postgresConfiguration, ConnectionFactory connectionFactory) {
        if (postgresConfiguration.rowLevelSecurityEnabled()) {
            LOGGER.info("PostgreSQL row level security enabled");
            LOGGER.info("Implementation for PostgreSQL connection factory: {}", DomainImplPostgresConnectionFactory.class.getName());
            return new DomainImplPostgresConnectionFactory(connectionFactory);
        }
        LOGGER.info("Implementation for PostgreSQL connection factory: {}", SinglePostgresConnectionFactory.class.getName());
        return new SinglePostgresConnectionFactory(Mono.from(connectionFactory.create()).block());
    }
```
If RLS enabled, we would use 1 connection per domain.
If RLS disabled, we would just use pool everywhere.

Then the connection factory (pool or not) would be backed by the `JamesPostgresConnectionFactory`, we do not need to care if this is POOL bean or not at the DAO layer.

WDYT?